### ISSUE TITLE
The output of `os.execute` is different from the execution output of the terminal

### ISSUE THREAD

**Issue by ShenTengTu** [vlang/v#10680]
**V version:** V 0.2.2 6dde9f7
**OS:**  Ubuntu 18.04, coreutils 8.28

**What did you do?**
Described in https://github.com/vlang/coreutils/pull/5#issuecomment-874809263

**What did you expect to see?**
In terminal, `printenv --null LANGUAGE LANG` output like following:
```
zh_TW:zhzh_TW.UTF-8
```

Use `os.execute` to execute the command:
```v
result_origin := os.execute('printenv --null LANGUAGE LANG')
println(result_origin.output)
```
the output must be `zh_TW:zhzh_TW.UTF-8`

**What did you see instead?**
the output  is `zh_TW:zh`

**Comment by shadowninja55** [comment]
sometimes `v run` does stuff, could you try compiling it and then running it manually?

**Comment by dumblob** [comment]
Hm, maybe (guessing without looking at the code) because V is incorrectly converting the pipe output to a C string (which is `\0` terminated) instead of converting it to *V* string (which doesn't have this restriction if I'm not mistaken).

**Comment by shadowninja55** [comment]
> Hm, maybe (guessing without looking at the code) because V is incorrectly converting the pipe output to a C string (which is `\0` terminated) instead of converting it to _V_ string (which doesn't have this restriction if I'm not mistaken).

that's quite possible, good thought.

**Comment by ShenTengTu** [comment]
> sometimes `v run` does stuff, could you try compiling it and then running it manually?

This problem still occurs in the binary version.

**Comment by spytheman** [comment]
Should be fixed in 20d63de.
